source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -48,20 +48,19 @@ UdpStatsdSink::UdpStatsdSink(ThreadLocal::SlotAllocator& tls,
   });
 }
 
-void UdpStatsdSink::flush(Stats::Source& source) {
+void UdpStatsdSink::flush(Stats::MetricSnapshot& snapshot) {
   Writer& writer = tls_->getTyped<Writer>();
-  for (const Stats::CounterSharedPtr& counter : source.cachedCounters()) {
-    if (counter->used()) {
-      uint64_t delta = counter->latch();
-      writer.write(fmt::format("{}.{}:{}|c{}", prefix_, getName(*counter), delta,
-                               buildTagStr(counter->tags())));
+  for (const auto& counter : snapshot.counters()) {
+    if (counter.counter_.get().used()) {
```

Comment:
counter.counter_->used() does not work here?

Reply:
`counter_` is a `reference_wrapper`